fastapi/src/ksi_oidc_fastapi/auth_middleware.py

```diff
@@ -0,0 +1,263 @@
+import logging
+from typing import Optional, Callable, Awaitable, Dict, List
+from fastapi import Request, Response, status
+from fastapi.responses import RedirectResponse
+from starlette.middleware.base import BaseHTTPMiddleware
+from fastapi.responses import JSONResponse
+from datetime import UTC, datetime, timedelta
+
+from ksi_oidc_common.tokens import AccessTokenClaims, Tokens
+from ksi_oidc_common.errors import OidcProviderError
+
+from .oidc_client import get_oidc_client
+
+from .models import Role
+from .session_manager import session_manager
+
+logger = logging.getLogger(__name__)
+
+
+class AuthMiddleware(BaseHTTPMiddleware):
+    """
+    FastAPI middleware for handling OIDC authentication sessions.
+    
+    This middleware:
+    1. Extracts session key from cookies
+    2. Loads session data and stores it in request.state.session_key
+    3. Provides helper properties for checking authentication
+    """
+    
+    def __init__(
+        self, 
+        app,
+        user_repository_instance : object,
+        route_configuration : Dict[Role, List[str]],
+        session_cookie_name: str = "session_id",
+        session_cookie_httponly: bool = True,
```

Comment:
Why is this even configurable? There should never be a need to access this cookie from JavaScript code and setting it as `HttpOnly` (slightly) reduces the risks associated with XSS attacks.